extensions/src/main/java/org/pentaho/platform/plugin/services/security/userrole/oauth/PentahoOAuthProviderFactory.java

```diff
@@ -0,0 +1,52 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.platform.plugin.services.security.userrole.oauth;
+
+import org.apache.tika.utils.StringUtils;
+
+public class PentahoOAuthProviderFactory {
+
+  PentahoOAuthAzureHandler pentahoOAuthAzureHandler;
+
+  PentahoOAuthOktaHandler pentahoOAuthOktaHandler;
+
+  public void setPentahoOAuthAzureHandler( PentahoOAuthAzureHandler pentahoOAuthAzureHandler ) {
+    this.pentahoOAuthAzureHandler = pentahoOAuthAzureHandler;
+  }
+
+  public void setPentahoOAuthOktaHandler( PentahoOAuthOktaHandler pentahoOAuthOktaHandler ) {
+    this.pentahoOAuthOktaHandler = pentahoOAuthOktaHandler;
+  }
+
+  /**
+   * Returns the appropriate IPentahoOAuthHandler instance based on the registrationId.
+   *
+   * @param registrationId The registration ID to determine which handler to return.
+   * @return The corresponding IPentahoOAuthHandler instance, or null if no match is found.
+   */
+  public IPentahoOAuthHandler getInstance( String registrationId ) {
```

Comment:
I feel you should either use enums or handle for insensitivity. 
`PentahoOAuthProviderFactory.java#getInsance(registrationId)`